theano/scan_module/scan.py

```diff
@@ -990,6 +992,10 @@ def wrap_into_list(x):
     info['as_while'] = as_while
     info['profile'] = profile
     info['allow_gc'] = allow_gc
+    info['cache_grad'] = cache_grad
+    if cache_grad:
+        warnings.warn('cache_grad works only when Scan Op is associated'
```

Comment:
I would make this warning more threatening. Something like:

Are are using Scan cache_grad. This is dangerous and can lead to WRONG results as it only work in some condition. This include that it work only with a single cost. If all conditions aren't meet, no error are raised, but wrong result are returned!